src/emulation/executable_mock.rs

```diff
@@ -15,7 +15,7 @@ pub fn create_mock_executable(context: &Context, config: Config) -> R<Vec<u8>> {
     let mut result = b"#!".to_vec();
     result.append(
         &mut context
-            .check_protocols_executable
+            .check_protocols_program
```

Comment:
because we might not be testing the behaviour of an executable if the interpreter is specified, this has been renamed.